File: tugas_pratikum/game_tebak_angka.py
import random

def play():
    while True:
        print("\n=== GAME TEBAK ANGKA ===")
        print("1. Mudah (1-50, 10 percobaan)")
        print("2. Sedang (1-100, 7 percobaan)")
        print("3. Sulit (1-500, 5 percobaan)")

        pilihan = input("Pilih level (1/2/3): ")

        if pilihan == "1":
            max_num = 50
            max_attempts = 10
        elif pilihan == "2":
            max_num = 100
            max_attempts = 7
        elif pilihan == "3":
            max_num = 500
            max_attempts = 5
        else:
            print("Pilihan tidak valid, otomatis level Sedang.")
            max_num = 100
            max_attempts = 7

        secret = random.randint(1, max_num)
        attempts = 0

        print(f"\nTebak angka antara 1 sampai {max_num}")

        while attempts < max_attempts:
            try:
                guess = int(input("Masukkan tebakan: "))
            except ValueError:
                print("Input harus angka!")
                continue

            attempts += 1

            if guess == secret:
                print(f"Benar! Kamu menang dalam {attempts} percobaan.")
                break

            if guess < secret:
                print("Terlalu kecil!")
            else:
                print("Terlalu besar!")

            # Hint panas / dingin
            if abs(guess - secret) <= 5:
                print("Panas! Sudah dekat.")
            else:
                print("Dingin! Masih jauh.")

            print("Sisa percobaan:", max_attempts - attempts)

        else:
            print(f"Game Over! Angka yang benar adalah {secret}")

        ulang = input("\nMain lagi? (y/n): ").lower()
        if ulang != 'y':
            print("Terima kasih sudah bermain!")
            break

if __name__ == "__main__":
    play()

File: tugas_pratikum/game_tebak_angka.c
// game_tebak_angka_upgrade.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int min = 1, max, secret, guess;
    int attempts, max_attempts;
    char playAgain;

    srand(time(NULL));

    do {
        int level;

        printf("\n=== GAME TEBAK ANGKA ===\n");
        printf("Pilih level:\n");
        printf("1. Mudah (1-50, 10 percobaan)\n");
        printf("2. Sedang (1-100, 7 percobaan)\n");
        printf("3. Sulit (1-500, 5 percobaan)\n");
        printf("Pilihan: ");
        scanf("%d", &level);

        switch(level) {
            case 1:
                max = 50;
                max_attempts = 10;
                break;
            case 2:
                max = 100;
                max_attempts = 7;
                break;
            case 3:
                max = 500;
                max_attempts = 5;
                break;
            default:
                printf("Level tidak valid, otomatis Sedang.\n");
                max = 100;
                max_attempts = 7;
        }

        secret = rand() % max + 1;
        attempts = 0;

        printf("\nTebak angka antara %d - %d\n", min, max);

        while(attempts < max_attempts) {
            printf("Masukkan tebakan: ");
            scanf("%d", &guess);

            attempts++;

            if(guess == secret) {
                printf("Benar! Kamu menang dalam %d percobaan.\n", attempts);
                break;
            }

            if(guess < secret)
                printf("Terlalu kecil!\n");
            else
                printf("Terlalu besar!\n");

            // Petunjuk panas/dingin
            if(abs(guess - secret) <= 5)
                printf("Panas! Sudah dekat.\n");
            else
                printf("Dingin! Masih jauh.\n");

            printf("Sisa percobaan: %d\n", max_attempts - attempts);
        }

        if(guess != secret)
            printf("Game Over! Angka yang benar: %d\n", secret);

        printf("\nMain lagi? (y/n): ");
        scanf(" %c", &playAgain);

    } while(playAgain == 'y' || playAgain == 'Y');

    printf("Terima kasih sudah bermain!\n");

    return 0;
}